src/context.rs

```diff
@@ -708,27 +709,53 @@ impl DefaultSeafowlContext {
         // On the other hand that would complicate etag testing logic.
         let table_uuid = get_uuid();
         let table_log_store = self.internal_object_store.get_log_store(table_uuid);
+        let table_prefix = self.internal_object_store.table_prefix(table_uuid);
 
         // NB: there's also a uuid generated below for table's `DeltaTableMetaData::id`, so it would
-        // be nice if those two could match
-        let table = Arc::new(
-            CreateBuilder::new()
-                .with_log_store(table_log_store)
-                .with_table_name(&*table_name)
-                .with_columns(delta_schema.fields().clone())
-                .with_comment(format!(
-                    "Created by Seafowl version {}",
-                    env!("CARGO_PKG_VERSION")
-                ))
-                .await?,
-        );
+        // be nice if those two could match somehow
+        let table = Arc::new(match details {
+            CreateDeltaTableDetails::WithSchema(schema) => {
+                let delta_schema = DeltaSchema::try_from(&schema)?;
+
+                CreateBuilder::new()
+                    .with_log_store(table_log_store)
+                    .with_table_name(&*table_name)
+                    .with_columns(delta_schema.fields().clone())
+                    .with_comment(format!(
+                        "Created by Seafowl {}",
+                        env!("CARGO_PKG_VERSION")
+                    ))
+                    .await?
+            }
+            CreateDeltaTableDetails::FromFiles(path) => {
+                // Copy the files over to the table root location
+                self.internal_object_store
+                    .copy_in_prefix(&path, &table_prefix)
+                    .await?;
+
+                // Now convert them to a Delta table
+                ConvertToDeltaBuilder::new()
+                    .with_log_store(table_log_store)
+                    .with_table_name(&*table_name)
+                    .with_comment(format!(
+                        "Converted by Seafowl {}",
+                        env!("CARGO_PKG_VERSION")
+                    ))
+                    .await?
```

Comment:
I think we do want to support completely-zero copy, but that means that either the source files have to be in a UUID-named directory inside of a prefix, or we have to track the full path for each table in the catalog (some intersection here with being able to persist external tables, including external Delta tables, probably an extension of https://github.com/splitgraph/seafowl/pull/472 where we assume all tables have the same prefix), or we do a destructive convert-to-delta and _move_ all Parquet files to a UUID-named directory.

Reply:
> we have to track the full path for each table in the catalog

Yeah, this makes the most sense to me going forward, but we'll need to think about the new catalog schema for that.

There's also a possibility of optionally outsourcing this info (completely or partially) to a 3rd party data catalog service, e.g. see here for how delta-rs envisions that: https://github.com/delta-io/delta-rs/blob/main/crates/deltalake-core/src/data_catalog/glue/mod.rs

For now though, I'm going to go with zero-copy, and assume the files are stored at exactly the right location (bucket + optional prefix + new table UUID).